src/CRUD.js

```diff
@@ -1,335 +1,308 @@
- * @returns {Array}      - collection of items matching the search
- */
-Database.prototype.find = function ( obj ) {
-  if (typeof obj === 'undefined') {
-    return this.findAll();
+
+  /**
+   * Initialization method
+   * @private
+   */
+  _initialize () {
+    // Load existing entries
+    this.data = this._load()
+    // Set internal id
+    this.id = Math.max(...this.data, 0)
   }
 
-  var keys = this.getKeys(obj),
-      filtered = [],
-      collection;
+  /**
+   * Finding entries
+   * @param   {Object} obj - the object of properties/values to look for
+   * @returns {Array}      - collection of items matching the search
+   */
+  find (obj = {}) {
+    // Grab the indexed and unindexed keys from search object
+    let keys = {
+      indexed: {},
+      unindexed: {}
+    }
 
-  for (var property in keys[0]) {
-    filtered.push(this.conf.driver.getItem(property + ':' + keys[0][property]) || false);
-  }
+    for (let property in obj) {
+      let stack = this.conf.indexedKeys.includes(property) ? 'indexed' : 'unindexed'
+      keys[stack][property] = obj[property]
+    }
 
-  if (filtered.length === 0) {
-    collection = this.data;
-  } else if (filtered.length === 1) {
-    collection = filtered[0];
-  } else {
-    collection = intersect.apply(this, filtered);
+    // Speed search results
+    let results = Object.keys(keys.indexed).map(prop =>
+      this.conf.driver.getItem(prop + ':' + keys.indexed[prop])
+    )
+
+    let collection = !results.length
+      ? this.data
+      : results.length === 1
+        ? results
+        : intersect(...results)
+
+    // Filtering by unindexed keys
+    return collection
+      .map(this.conf.driver.getItem, this.conf.driver)
+      .filter(entry =>
+        Object.keys(keys.unindexed)
+          .every(key => entry[key] === keys.unindexed[key])
+      )
   }
 
-  // Filtering by unindexed keys
-  return this.filter(this.select(collection), keys[1]);
-};
-
-/**
- * Returning all entries from database
- * @returns {Array} - collection of entries
- */
-Database.prototype.findAll = function () {
-  return this.select(this.data);
-};
-
-/**
- * Dissociate indexed from unindexed keys from an object
- * @param   {Object} obj - object to parse
- * @returns {Array}      - array of indexed keys and unindexed keys
- */
-Database.prototype.getKeys = function ( obj ) {
-  var index, keys = [];
-  keys[0] = {}; // indexed properties
-  keys[1] = {}; // unindexed properties
-
-  for(var property in obj) {
-    index = this.conf.indexedKeys.indexOf(property) !== -1 ? 0 : 1;
-    keys[index][property] = obj[property];
-  }
+  /**
+   * Inserting an entry
+   * @param   {Array|Object} arg - entry or array of entries to insert
+   * @returns {Number}           - unique key of the document
+   */
+  insert (arg) {
+    // Go recursive if it is an array, inserting several entries at once
+    if (Array.isArray(arg)) {
+      return arg.forEach(this.insert, this)
```

Comment:
Performance reasons. Babel compiles `::this.insert` into `this.insert.bind(this)`, not using `thisArg` from `forEach(..)`, which is slower.